atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -121,6 +121,13 @@ public void run() {
             Thread.sleep(getBackoffTimeWhenSweepHasNotRun());
             log.info("Starting background sweeper.");
             while (true) {
+                // InterruptedException might be wrapped in RuntimeException (i.e. AtlasDbDependencyException),
+                // which would be caught downstream.
+                // Therefore we need to actively check if the thread has been interrupted here, and exit sweep if so.
+                if (Thread.currentThread().isInterrupted()) {
+                    break;
```

Comment:
Can we rethrow the InterruptedException instead?  This way we hit the catch block below and move to SHUTDOWN state.  Useful for metrics and monitoring

Reply:
Done.